core/bodydb.go

```diff
@@ -69,10 +72,16 @@ func (bc *BodyDb) Append(batch ethdb.Batch, block *types.Block, newInboundEtxs t
 	defer bc.chainmu.Unlock()
 
 	stateApply := time.Now()
-	// Process our block
-	logs, err := bc.processor.Apply(batch, block, newInboundEtxs)
-	if err != nil {
-		return nil, err
+	nodeCtx := common.NodeLocation.Context()
+	var logs []*types.Log
+	var err error
+	if nodeCtx == common.ZONE_CTX {
+		// Process our block
+		logs, err = bc.processor.Apply(batch, block, newInboundEtxs)
+		if err != nil {
+			return nil, err
+		}
+		rawdb.WriteTxLookupEntriesByBlock(batch, block)
```

Comment:
This write will happen even if `block.Hash() != block.Header().Hash()` two lines below.

I'm not sure the origin of that check, or how it could fail, so maybe we should just remove it?

Reply:
Good point. Perhaps this check should be removed?

Reply:
Removed the check  completely 
resolves #825